mmf/datasets/mmf_dataset.py

```diff
@@ -50,6 +52,37 @@ def build_image_db(self):
         image_path = self._get_path_based_on_index(self.config, "images", self._index)
         return ImageDatabase(self.config, image_path, annotation_db=self.annotation_db)
 
+    def format_for_caption_generation(self, report):
```

Comment:
Let's not add it here in MMFDataset. This should go to the individual datasets that require caption generation. For example for Hateful Memes, I would suggest add a separate dataset class for HM caption and add `format_for_prediction` in that which will generate caption predictions.

Reply:
So I was thinking of a use-case where any dataset registered to MMF can directly use the `butd` model for caption generation, or any other subsequent models for caption generation. So something like this would work immediately, as long as the model configs point to the butd model and the dataset configs point to that dataset.

`model=butd`
`dataset=<any-mmf-registered-dataset>`